hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/multitenant/RangerAccessPolicy.java

```diff
@@ -0,0 +1,250 @@
+      resourceItems.append("\"]," +
+          "\"isRecursive\":true," +
+          "\"isExcludes\":false" +
+          "}");
+    }
+    resourceItems.append("},");
+    return resourceItems.toString();
+  }
+
+  private String createRangerPolicyItems() throws IOException {
+    StringBuilder policyItems = new StringBuilder();
+    policyItems.append("\"policyItems\":[");
+    int mapRemainingSize = policyMap.size();
+    for (Map.Entry<String, List<AccessPolicyElem>> mapElem : policyMap
+        .entrySet()) {
+      mapRemainingSize--;
+      List<AccessPolicyElem> list = mapElem.getValue();
+      if (list.isEmpty()) {
+        continue;
+      }
+      policyItems.append("{");
+      if (list.get(0).getPrincipal().getUserPrincipalType()
+          == GROUP_PRINCIPAL) {
+        policyItems.append("\"groups\":[\"" + mapElem.getKey() + "\"],");
+      } else {
+        policyItems.append("\"users\":[\"" + mapElem.getKey() + "\"],");
+      }
+      policyItems.append("\"accesses\":[");
+      Iterator<AccessPolicyElem> iter = list.iterator();
+      while (iter.hasNext()) {
+        AccessPolicyElem elem = iter.next();
+        policyItems.append("{");
+        policyItems.append("\"type\":\"");
+        policyItems.append(getRangerAclString(elem.getAclType()));
+        policyItems.append("\",");
+        if (elem.getAccessGrantType() == AccessGrantType.ALLOW) {
+          policyItems.append("\"isAllowed\":true");
+        } else {
+          policyItems.append("\"isDenied\":true");
+        }
+        policyItems.append("}");
+        if (iter.hasNext()) {
+          policyItems.append(",");
+        }
+      }
+      policyItems.append("]");
+      policyItems.append("}");
+      if (mapRemainingSize > 0) {
+        policyItems.append(",");
+      }
+    }
+    policyItems.append("],");
+    return policyItems.toString();
+  }
+
+  private String getRangerAclString(ACLType aclType) throws IOException {
+    switch (aclType) {
+    case ALL:
+      return "All";
+    case LIST:
+      return "List";
+    case READ:
+      return "Read";
+    case WRITE:
+      return "Write";
+    case CREATE:
+      return "Create";
+    case DELETE:
+      return "Delete";
+    case READ_ACL:
+      return "Read_ACL";
+    case WRITE_ACL:
+      return "Write_ACL";
+    case NONE:
+      return "";
+    default:
+      throw new IOException("Unknown ACLType");
+    }
+  }
+
+  private void updatePolicyJsonString() throws Exception {
+    policyJsonString =
+        "{\"policyType\":\"0\"," + "\"name\":\"" + policyName + "\","
+            + "\"isEnabled\":true," + "\"policyPriority\":0,"
+            + "\"policyLabels\":[]," + "\"description\":\"\","
+            + "\"isAuditEnabled\":true," + createRangerResourceItems()
+            + "\"isDenyAllElse\":false," + createRangerPolicyItems()
+            + "\"allowExceptions\":[]," + "\"denyPolicyItems\":[],"
+            + "\"denyExceptions\":[]," + "\"service\":\"cm_ozone\"" + "}";
```

Comment:
In the future, service name "cm_ozone" may be something that Ozone needs to figure out from Ranger.

Reply:
Yes.